src/com/github/aroq/drupipe/actions/Shell.groovy

```diff
@@ -13,7 +13,7 @@ class Shell extends BaseAction {
     def DrupipeAction action
 
     def execute() {
-        script.drupipeShell(action.params.shellCommand, context)
+        def result = script.drupipeShell(action.params.shellCommand, this.context.clone() << [drupipeShellReturnStdout: true])
```

Comment:
Are we really always need it? Keep in mind that unbuffered output will not be possible in this case (for logs).